src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -732,7 +737,19 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
             petType = HUNTER_PET;
             m_unitTypeMask |= UNIT_MASK_HUNTER_PET;
         }
-        else
+
+        switch (sScriptMgr->OverridePetType(m_owner->ToPlayer(), cinfo))
+        {
+            case SUMMON_PET:
+                petType = SUMMON_PET;
+                break;
+            case HUNTER_PET:
+                petType = HUNTER_PET;
+                m_unitTypeMask |= UNIT_MASK_HUNTER_PET;
+                break;
+        }
+
+        if (petType != SUMMON_PET && petType != HUNTER_PET)
             sLog->outError("Unknown type pet %u is summoned by player class %u", GetEntry(), m_owner->getClass());
```

Comment:
why not put this last line inside a `default` of the above `switch` ?

Reply:
Because the petType can be set before the switch statement and the switch statement just handles the hook.